core/primitives/src/runtime/config.rs

```diff
@@ -51,30 +47,54 @@ impl RuntimeConfig {
             account_creation_config: AccountCreationConfig::default(),
         }
     }
+}
+
+/// An actual runtime configuration the node will use.
+///
+/// This is constructed from the runtime configuration given in the genesis file
+/// but i) may have it’s `max_gas_burnt_view` limit adjusted and ii) provides
+/// a method which returns configuration with adjustments done through protocol
+/// version upgrades.
+pub struct ActualRuntimeConfig {
+    /// The runtime configuration taken from the genesis file but with possibly
+    /// modified `max_gas_burnt_view` limit.
+    runtime_config: Arc<RuntimeConfig>,
+
+    /// The runtime configuration with lower storage cost adjustment applied.
+    with_lower_storage_cost: Arc<RuntimeConfig>,
```

Comment:
Does this mean that every time we change the runtime config through a protocol change we will need to introduce a new field? Is that a desirable way to do it?

Reply:
> Does this mean that every time we change the runtime config through a protocol change we will need to introduce a new field?

Yes. Note that this commit doesn’t change this. This is the same behaviour as before.

> Is that a desirable way to do it?

Maybe not, but if we want to have ability to get the configuration for any possible protocol version, we either need to store the configs or build the configs on the fly each time a config is requested. If we want to change how this is handled I can certainly do it but in a separate commit since scope of this one is only to get rid of the global.